QuasarHelios2015/src/org/team1540/quasarhelios/ControlInterface.java

```diff
@@ -35,23 +39,65 @@ public static void setupJoysticks() {
+        FloatInput povAngle = Igneous.joystick2.getPOVAngleSource(1);
+        EventInput up = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, -0.1f, 0.1f), true, povPressed);
+        EventInput down = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 179.9f, 180.1f), true, povPressed);
+        EventInput left = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 269.9f, 270.1f), true, povPressed);
+        EventInput right = EventMixing.filterEvent(FloatMixing.floatIsInRange(povAngle, 89.9f, 90.1f), true, povPressed);
+        
+        up.send(() -> {
+            if (Rollers.direction.get()) {
+                Rollers.running.set(true);
+            } else {
+                Rollers.running.set(!Rollers.running.get());
+            }
+            
+            Rollers.direction.set(false);
+        });
+        
+        down.send(() -> {
+            if (!Rollers.direction.get()) {
+                Rollers.running.set(true);
+            } else {
+                Rollers.running.set(!Rollers.running.get());
+            }
+            
+            Rollers.direction.set(true);
+        });
+        
+        Rollers.open.toggleWhen(left);
 
-        Clamp.heightInput = Igneous.joystick2.getAxisSource(2);
-        Clamp.openInput = Igneous.joystick2.getButtonSource(8);
+        FloatInput cutoffRollers = mainTuning.getFloat("roller-override-threshold", 0.3f);
+        BooleanInputPoll overrideRollers = Igneous.joystick2.getButtonChannel(6);
 
-        QuasarHelios.autoLoader.toggleWhen(Igneous.joystick2.getButtonSource(9));
-        QuasarHelios.autoEjector.toggleWhen(Igneous.joystick2.getButtonSource(10));
+        BooleanMixing.pumpWhen(QuasarHelios.globalControl, BooleanMixing.andBooleans(overrideRollers, FloatMixing.floatIsAtMost(
+                Igneous.joystick2.getAxisChannel(1), cutoffRollers)), Rollers.leftPneumaticOverride);
+
+        axis2.send(Rollers.leftRollerOverride);
+
+        BooleanMixing.pumpWhen(QuasarHelios.globalControl, BooleanMixing.andBooleans(overrideRollers, FloatMixing.floatIsAtLeast(
+                Igneous.joystick2.getAxisChannel(5), FloatMixing.negate(cutoffRollers))), Rollers.rightPneumaticOverride);
+
+        
+        FloatInput axis6 = Igneous.joystick2.getAxisSource(6);
+        axis6.send(Rollers.rightRollerOverride);
+
+        FloatMixing.pumpWhen(EventMixing.filterEvent(overrideRollers, false, QuasarHelios.globalControl), axis6, Clamp.height);
+        Clamp.open.toggleWhen(Igneous.joystick2.getButtonSource(3));
+
+        FloatInput cutoffAuto = mainTuning.getFloat("autoloader-button-threshold", 0.5f);
+
+        BooleanMixing.pumpWhen(QuasarHelios.globalControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(3), cutoffAuto), QuasarHelios.autoEjector);
+        BooleanMixing.pumpWhen(QuasarHelios.globalControl, FloatMixing.floatIsAtLeast(Igneous.joystick2.getAxisSource(4), cutoffAuto), QuasarHelios.autoLoader);
```

Comment:
I think that this method is getting a bit long. Can we cut it up to make what's happening clearer?